backend/app/actions/infer_action.py

```diff
@@ -47,15 +49,30 @@
+# PROMPT_SQL_TRANSLATION = """
+# Given the identified fields from the user's query that require SQL translation, generate appropriate SQL WHERE clause conditions. This task focuses on converting these field references into precise SQL queries.
+
+# Identified fields: {identified_fields}
+# Current user query: "{cur_query}"
+
+# For each identified metadata field, create the corresponding SQL WHERE clause condition. Ensure the translation reflects any specific conditions mentioned in the query, such as precise dates, numerical ranges, or other descriptive qualifiers.
+
+# The temporal granularity should be chosen from Year, Quarter, Month, Week, Day, Hour, Minute, or Second. The geographical granularity should be chosen from Continent, Country, State/Province, County/District, City, or Zip Code/Postal Code.
+
+# For example, if the fields 'Temporal Granularity' and 'Geographic Granularity' are identified in user's query 'I only want data in United States after 2020', the WHERE clause for temporal_granularity might be "= 'year'", and the clause for geographic_granularity might be "= 'country'".
+# """
+
 PROMPT_SQL_TRANSLATION = """
-Given the identified fields from the user's query that require SQL translation, generate appropriate SQL WHERE clause conditions. This task focuses on converting these field references into precise SQL queries.
+Given the identified fields from the user's query requiring SQL translation, generate appropriate SQL WHERE clause conditions. This task involves converting these field references into precise SQL queries, adhering strictly to predefined granularity levels.
 
 Identified fields: {identified_fields}
 Current user query: "{cur_query}"
 
-For each identified metadata field, create the corresponding SQL WHERE clause condition. Ensure the translation reflects any specific conditions mentioned in the query, such as precise dates, numerical ranges, or other descriptive qualifiers.
+For each identified metadata field, create the corresponding SQL WHERE clause condition to exactly match the predefined granularity levels.
+
+The temporal granularity should always be referenced specifically as one of the following: Year, Quarter, Month, Week, Day, Hour, Minute, or Second. The geographical granularity should be one of the following: Continent, Country, State/Province, County/District, City, or Zip Code/Postal Code.
 
-For example, if the fields 'Temporal Granularity' and 'Geographic Granularity' are identified in user's query 'I only want data in United States after 2020', the WHERE clause for temporal_granularity might be ">= '2020-01-01'", and the clause for geographic_granularity might be "= 'US'".
+For example, if the fields 'Temporal Granularity' and 'Geographic Granularity' are identified in the user's query 'I only want data in the United States after 2020', the WHERE clause for temporal granularity might be "= 'year'" and for geographic granularity should be "= 'country'".
```

Comment:
Is this example accurate? I guess we aim to facilitate metadata queries that express "the dataset should provide annual data on country-level".